File: Module3/Lab3_12_UITests-student/app/src/main/java/ru/myitschool/lab23/MainActivity.java
package ru.myitschool.lab23;


import android.app.job.JobInfo;
import android.app.job.JobScheduler;
import android.content.ComponentName;
import android.content.Context;
import android.os.Bundle;
import android.os.PersistableBundle;
import android.widget.EditText;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    protected final static String keyText = "Property Key";
    private EditText propertyText;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        propertyText = findViewById(R.id.property_text);

        findViewById(R.id.submit_button).setOnClickListener(v -> {
            JobScheduler jobScheduler = (JobScheduler) getSystemService(Context.JOB_SCHEDULER_SERVICE);
            PersistableBundle textBundle = new PersistableBundle();
            textBundle.putString(keyText, propertyText.getText().toString());
            JobInfo jobInfo = new JobInfo.Builder(13, new ComponentName(this, DownloadService.class)).setExtras(textBundle).build();
            jobScheduler.schedule(jobInfo);
        });
    }
}

File: Module3/PR3.2/app/src/main/java/ru/adwantay/lab/MainActivity.kt
package ru.adwantay.lab

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    private lateinit var eAddress: EditText
    private lateinit var eTheme: EditText
    private lateinit var eLetter: EditText
    private lateinit var btnSend: Button
    private lateinit var btnCall: Button
    private lateinit var ePhNumber: EditText

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btnSend = findViewById(R.id.bsend)
        btnCall = findViewById(R.id.bcall)
        eAddress = findViewById(R.id.eadress)
        eTheme = findViewById(R.id.etitle)
        eLetter = findViewById(R.id.etext)
        ePhNumber = findViewById(R.id.etelno)


        btnSend.setOnClickListener {
            if (eAddress.text.isNotEmpty() && eTheme.text.isNotEmpty() && eLetter.text.isNotEmpty()) {
                val intent = Intent(Intent.ACTION_SENDTO)
                intent.apply {
                    type = "text/plain";
                    putExtra(Intent.EXTRA_EMAIL, eAddress.text.toString())
                    putExtra(Intent.EXTRA_SUBJECT, eTheme.text.toString())
                    putExtra(Intent.EXTRA_TEXT, eLetter.text.toString())
                }
                startActivity(intent)
            }
        }

        btnCall.setOnClickListener{
            if(ePhNumber.text.isNotEmpty()){
                val intent = Intent(Intent.ACTION_VIEW, Uri.parse("tel:${ePhNumber.text}"))
                startActivity(intent)
            }
        }


    }



}
